R/plumber.R

```diff
@@ -893,6 +830,33 @@ plumber <- R6Class(
     setErrorHandler = function(fun){
       private$errorHandler <- fun
     },
+    #' @description Add a function to customize what is returned in `$apiSpec()`.
+    #' @param api_fun This function should accept an OpenAPI Specification autogenerated by `plumber`.  It should return a list object in valid OpenAPI Specification format. This will not be validated.
+    setApiHandler = function(api_fun = force) {
+      # allow null values
+      if (!is.null(api_fun)) {
+        stopifnot(is.function(api_fun))
+      }
+      private$apiHandler <- api_fun
+    },
+    #' @description Set UI to use for API
+    #' @param ui a character value or a logical value. Default to `plumber.ui` option value.
+    #' @param ... Other params to be passed to ui functions.
+    setUI = function(ui = getOption("plumber.ui", TRUE), ...) {
+      stopifnot(isTRUE(length(ui) == 1L))
+      stopifnot(is.logical(ui) || is.character(ui))
+      if (isTRUE(ui) || is.character(ui)) {
```

Comment:
Must set a value to a character string. For now, let's default to `"swagger"`

```suggestion
      if (isTRUE(ui)) {
        ui <- "swagger"
      }
      if (is.character(ui)) {
```